pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformer.java

```diff
@@ -279,6 +295,31 @@ protected void flattenMap(GenericRow record, List<String> columns) {
     }
   }
 
+  /**
+   * Loops through all columns and renames the column's prefix with the corresponding replacement if the prefix matches.
+   */
+  @VisibleForTesting
+  protected void renamePrefixes(GenericRow record) {
+    if (_prefixesToRename.isEmpty()) {
+      return;
+    }
+    List<String> fields = new ArrayList<>(record.getFieldToValueMap().keySet());
```

Comment:
Avoid creating this extra list

Reply:
Ah actually, we need this list to avoid ConcurrentModificationException